### PR TITLE
Bug/118 supported actions explaintypes json not considered

### LINKED ISSUE #118: [BUG] Supported actions (explainTypes --json) not considered
Copy and Copy & Deploy menu options should only be displayed to the metadata types that are can be deployed by checking the json retrieved by explain types

 - if metadata can be deployed -> shows menu with copy and copy & deploy
 - if metadata cannot be deployed -> shows an error  

retrieve -> retrieve
deploy -> create || update
copy -> create || update
copy & deploy --> create || update

### PR DESCRIPTION
**Pull request by anasilva105**
# PR details

## What changes did you make? (Give an overview)

- All the menu action (Copy, Deploy and Retrieve) now check if if they are supported in SFMC DevTools according to the following retrieved from the explainTypes command:
     - retrieve -> retrieve
     - deploy -> create || update
     - copy -> create || update
     - copy & deploy --> create || update 

- Notification Message is displayed when one or more metadata types is not supported
image

- closes #118

### CODE REVIEW

**Comment by JoernBerkefeld** [comment]
Could you Post a Screenshot of the notification that users get  please

**Comment by anasilva105** [comment]
> Could you Post a Screenshot of the notification that users get please

Added to the PR

File: src/devtools/main.ts [hunk 1]
```diff
@@ -362,12 +367,35 @@ async function initialize(): Promise<void>{
...
+    [unsupportedMtdtTypes]
+    .flat()
+    .forEach((metadataType: string) => {
+        log(
+            "info", 
+            `SFMC DevTools currently does not support ${action} for the metadata type: '${metadataType}'`
+        );
+    });
+    devtoolsContainers.modifyStatusBar("mcdev", "info");
+    editorInput.handleShowNotificationMessage("info", mainConfig.messages.unsupportedMetadataType, []);
```
Comment (JoernBerkefeld) [inline src/devtools/main.ts:394]:
```suggestion
    editorInput.handleShowNotificationMessage("error", mainConfig.messages.unsupportedMetadataType, []);
```

argument could be made for "warning" but I believe that this should be stronger and hence lean towards "error"

File: src/devtools/main.ts [hunk 2]
```diff
@@ -362,12 +367,35 @@ async function initialize(): Promise<void>{
...
+    ]: string[] = relativePath.split("/");
+    return { credentialName, businessUnit, metadataType, keys };
+}
+
+function logUnsupportedMtdtTypeNotification(action: string, unsupportedMtdtTypes: string | string[]){
+    [unsupportedMtdtTypes]
+    .flat()
+    .forEach((metadataType: string) => {
+        log(
+            "info",
```
Comment (JoernBerkefeld) [inline src/devtools/main.ts:389]:
```suggestion
            "error", 
```
should be an error, not info.

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]

**Comment by anasilva105** [comment]
@JoernBerkefeld what about the notification displayed should it be an error, warning or info ?

warning
image


error:
image


info:
image

**Comment by JoernBerkefeld** [comment]
i assumed my proposed changes would cascade down to that particular pop-up --> error.
that being said. can we do add a "more details" button instead of the directions in sentence #2?

**Comment by anasilva105** [comment]
@JoernBerkefeld updated solution to display a More Details button. Clicking on this button will open the extension Output Channel:

image

**Review by JoernBerkefeld — APPROVED** [review]

**Pull request merged**